grails-app/services/org/pih/warehouse/core/UserService.groovy

```diff
@@ -187,21 +187,23 @@ class UserService {
 
 
 
-
-    def findPersons(String query, Map params) {
-        def criteria = Person.createCriteria()
-        def results = criteria.list (params) {
-            or {
-                ilike("firstName", query)
-                ilike("lastName", query)
-                ilike("email", query)
+    def findPersons(String[] terms) {
+        def results = Person.createCriteria().list {
+
+            if (terms) {
+                terms.each { term ->
```

Comment:
I think there could be an ```and { ... }``` around the entire query as I showed in the pseudocode.

Reply:
Actuall, I just tested it myself and it looks like the AND is implied already.
{code}
SELECT 
    this_.id AS id65_0_,
    this_.version AS version65_0_,
    this_.date_created AS date3_65_0_,
    this_.email AS email65_0_,
    this_.first_name AS first5_65_0_,
    this_.last_name AS last6_65_0_,
    this_.last_updated AS last7_65_0_,
    this_.phone_number AS phone8_65_0_,
    this_1_.active AS active66_0_,
    this_1_.last_login_date AS last3_66_0_,
    this_1_.locale AS locale66_0_,
    this_1_.manager_id AS manager5_66_0_,
    this_1_.password AS password66_0_,
    this_1_.photo AS photo66_0_,
    this_1_.remember_last_location AS remember8_66_0_,
    this_1_.timezone AS timezone66_0_,
    this_1_.username AS username66_0_,
    this_1_.warehouse_id AS warehouse11_66_0_,
    CASE
        WHEN this_1_.id IS NOT NULL THEN 1
        WHEN this_.id IS NOT NULL THEN 0
    END AS clazz_0_
FROM
    person this_
        LEFT OUTER JOIN
    `user` this_1_ ON this_.id = this_1_.id
WHERE
    (
		(	
			LOWER(this_.first_name) LIKE '%jus%'
			OR LOWER(this_.last_name) LIKE '%jus%'
			OR LOWER(this_.email) LIKE '%jus%'
		)
		AND 
		(	
			LOWER(this_.first_name) LIKE '%mir%'
			OR LOWER(this_.last_name) LIKE '%mir%'
			OR LOWER(this_.email) LIKE '%mir%'
		)
	)
ORDER BY this_.last_name DESC
{code}

Reply:
@jmiranda so is there anything to add? I've testes it and it worked as it was supposed to.